diff --git a/wolfcrypt/src/wc_port.c b/wolfcrypt/src/wc_port.c
--- a/wolfcrypt/src/wc_port.c
+++ b/wolfcrypt/src/wc_port.c
@@ -421,6 +421,9 @@ int aarch64_use_sb = 0;
 WOLFSSL_ABI
 int wolfCrypt_Init(void)
 {
+#if defined(WOLFSSL_PALM_STATELESS_INIT)
+    return 0;
+#else
     int ret;
 #if defined(HAVE_THREAD_LS) && !defined(NO_THREAD_LS) && defined(__GNUC__)
     /* If thread-local storage is available, use it to prevent deadlock on
@@ -770,2 +773,3 @@ int wolfCrypt_Init(void)
     /* not reached */
+#endif
 }
@@ -793,4 +797,7 @@ long wolfCrypt_heap_peakBytes_checkpoint(void) {
 WOLFSSL_ABI
 int wolfCrypt_Cleanup(void)
 {
+#if defined(WOLFSSL_PALM_STATELESS_INIT)
+    return 0;
+#else
     int ret;
@@ -933,2 +940,3 @@ int wolfCrypt_Cleanup(void)
     /* not reached */
+#endif
 }
