diff --git a/wolfssl/ssl.h b/wolfssl/ssl.h
index 86478d3f9..63318c76f 100644
--- a/wolfssl/ssl.h
+++ b/wolfssl/ssl.h
@@ -4517,9 +4517,9 @@ enum {
 };

 WOLFSSL_ABI WOLFSSL_API int wolfSSL_UseSNI(WOLFSSL* ssl, unsigned char type,
-                                                   const void* data, unsigned short size);
+                                                   const void* data, word16 size);
 WOLFSSL_ABI WOLFSSL_API int wolfSSL_CTX_UseSNI(WOLFSSL_CTX* ctx, unsigned char type,
-                                                   const void* data, unsigned short size);
+                                                   const void* data, word16 size);

 #ifndef NO_WOLFSSL_SERVER
